pkg/kubelet/kubelet.go

```diff
@@ -1984,6 +1995,54 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	return false, nil
 }
 
+// waitEventsBySyncPod waits for PLEG to get events along with actions in SyncPod() and update the pod status in the cache.
+func (kl *Kubelet) waitEventsBySyncPod(pod *v1.Pod, result *kubecontainer.PodSyncResult, syncStartTime time.Time) {
+	resultsToWait := make([]*kubecontainer.SyncResult, 0, len(result.SyncResults))
+	for _, r := range result.SyncResults {
+		if r.Error == nil && (r.Action == kubecontainer.StartContainer || r.Action == kubecontainer.KillContainer) {
+			resultsToWait = append(resultsToWait, r)
+		}
+	}
+	if len(resultsToWait) == 0 {
+		return
+	}
+
+	lastGetTime := syncStartTime
+	maxWaitTime := kl.clock.Now().Add(maxWaitForEventsAfterSyncPod)
+wait_event:
```

Comment:
this is not great when we need labels to express some looping. Indicating that code is hard to read. Maybe refactor it